jar-tool/Main.hx

```diff
@@ -0,0 +1,108 @@
+package;
+
+import java.StdTypes.Int8;
+import java.NativeArray;
+import java.io.FileOutputStream;
+import java.io.File;
+import java.nio.file.Files;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes.Opcodes_Statics;
+import java.Lib.println;
+
+using StringTools;
+
+/**
+ * HOW TO USE:
+ *
+ * 1) Unzip JAR file into some folder. For example "jar_content";
+ * 2) Run `java -jar Main.jar jar_content`;
+ * 3) Pick up generated `hxjava-std.jar`.
+ */
+final class Main {
+	static function main():Void {
+		function process(dir:String) {
+			if (sys.FileSystem.exists(dir)) {
+				for (file in sys.FileSystem.readDirectory(dir)) {
+					final path = haxe.io.Path.join([dir, file]);
+					if (!sys.FileSystem.isDirectory(path)) {
+						if (path.endsWith(".class")) {
+							println('Process $path... ');
+							final obj:NativeArray<Int8> = Files.readAllBytes(new File(path).toPath());
+							final reader = new ClassReader(obj);
+							final writer = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
+							final visitor = new MyClassVisitor(writer);
+
+							reader.accept(visitor, 0);
+
+							final stream = new FileOutputStream(path);
+							stream.write(writer.toByteArray());
+							Sys.sleep(0.001);
+						}
+					} else {
+						process(haxe.io.Path.addTrailingSlash(path));
+					}
+				}
+			} else {
+				println('Wrong path. $dir is not exists');
+			}
+		}
+
+		final args:Array<String> = Sys.args();
+		if (args.length > 0) {
+			final dir = args[0];
+			process(dir);
+			println("\nPacking jar...\n");
+			if (Sys.command('jar', ["cvf", "hxjava-std.jar", "-C", dir, "."]) == 0) {
+				println("\nDone!\n");
+			} else {
+				println("Oops, something went wrong.");
+			}
+		}
+	}
+}
+
+final class MyClassVisitor extends ClassVisitor {
+	var isInterface:Bool = false;
+
+	public function new(visitor:ClassVisitor) {
+		super(Opcodes_Statics.ASM7, visitor);
+	}
+
+	@:overload
+	override function visit(version:Int, access:Int, name:String, signature:String, superName:String, interfaces:NativeArray<String>):Void {
+		cv.visit(version, access, name, signature, superName, interfaces);
+		isInterface = (access & Opcodes_Statics.ACC_INTERFACE) != 0;
+	}
+
+	@:overload
+	override function visitMethod(access:Int, name:String, desc:String, signature:String, exceptions:NativeArray<String>):MethodVisitor {
+		final mv:MethodVisitor = cv.visitMethod(access, name, desc, signature, exceptions);
+		if (!isInterface && mv != null) {
```

Comment:
Shouldn't this check something along the lines of "if there is any code" instead of explicitly checking for interfaces? There are abstract methods too where we shouldn't add any code.

Reply:
Yes, but I check it here:
https://github.com/dmitryhryppa/hxjava/blob/master/jar-tool/Main.hx#L31

Reply:
That's not what I mean. The condition should check if there is a [Code](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.3) attribute.

Reply:
Looks like `visitCode` only does anything if there is a `Code` attribute. Though in this case the interface check should be unnecessary, but that's not very important.

Reply:
Got it. I did a small cleanup based on your comments.